dev/test/intersection.test.ts

```diff
@@ -77,6 +83,32 @@ describe("intersection", () => {
                 object: { props: { a: "string", b: "boolean" } }
             })
         })
+        describe("Tuple and Array", () => {
+            it("string type", () => {
+                const t = type([["string", "string"], "&", "string[]"])
+                attest(t(["1", "1"]).data).snap(["1", "1"])
+                attest(t(["1", 1]).problems?.summary).snap(
+                    "Item at index 1 must be a string (was number)"
+                )
+            })
+            it("multiple types with union array", () => {
+                const t = type([["number", "string"], "&", "(string|number)[]"])
```

Comment:
Again, make the intersected array types more specific (e.g. `('one'|1)[]`) so we know that they were correctly merged.